### ISSUE TITLE
Add debug console

### ISSUE THREAD

**Issue by Kasterisk** [Revolutionary-Games/Thrive#3414]
We should add a debug/command console to the game that could handle command inputs and showing the game logs during runtime. This will be extremely helpful for playtesting and debugging where we could have easier access to many more cheats and debugging tools.

**Comment by hhyyrylainen** [comment]
I've added a dedicated folder for scenes that just load a specific stage in the game, it'd be pretty handy if the console could be used to switch to these scenes with a quick command to allow people playing a release version of Thrive to jump to a specific stage.

**Comment by hhyyrylainen** [comment]
Note that it seems there's currently no way to get current log output from Godot, so we couldn't show Godot messages in the console...

Relevant godot proposal:

https://github.com/godotengine/godot-proposals/issues/5937

**Comment by hhyyrylainen** [comment]
Also once this is done the testing wiki page needs to be updated to mention these new powerful tools: https://wiki.revolutionarygamesstudio.com/wiki/Testing

**Comment by hhyyrylainen** [comment]
This could be taken inspiration from: https://github.com/MASSHUU12/godot-yat (or maybe even used as-is if system commands can be disabled for more safety).

**Comment by xfractalino** [comment]
@hhyyrylainen Godot 4.5 introduced a [new Logging API](https://docs.godotengine.org/en/stable/tutorials/scripting/logging.html), which supports custom callbacks. Perhaps, it could be used to push the messages to an in-game Console.

**Comment by hhyyrylainen** [comment]
Yes, I believe so. I did actually write a feature with that logging API already for Thrive. That's how the new "unhandled error occurred" popup actually works and can detect errors.

**Comment by xfractalino** [comment]
Yeah, I see. I'm just now taking a look at the LogInterceptor class. I'll play with it a bit.

**Comment by hhyyrylainen** [comment]
Now that this is done I opened a follow up issue: https://github.com/Revolutionary-Games/Thrive/issues/6631
